AirCasting/Settings/BackendURLValidator.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  BackendURLValidator.swift
+//  URLValidation
+//
+//  Created by Pawel Gil on 25/06/2021.
+//
+
+import Foundation
+
+class BackendURLBuilder {
+    enum ValidationError: Error {
+        case invalidURL
+        case invalidPort
+        case noURL
+    }
+    
+    func createURL(url: String, port: String) throws -> URL? {
+        if url.count == 0 && port.count == 0 { return nil }
+        if url.count == 0 && port.count > 0 { throw ValidationError.noURL }
+        let url = appenURLWithSchemeIfNotPresent(url: url)
+        var components = try getComponents(url: url)
+        try validateHost(for: components)
+        components.port = try getPortNumber(from: port)
+        return components.url!
```

Comment:
even tho we know at this point this `components.url` really should be non-nil, let's just get rid of the bang here and do:
```Swift
return components.url
```
we won't harm any logic and perhaps avoid some unpredicted crashes here